### PR TITLE
Add GitHub Actions workflow to build the Docker image

### PR DESCRIPTION
**Pull request by psavery**
This will automatically build and push images to DockerHub with every merge.
    
We are currently building amd64 and arm64 images.

### CODE REVIEW

**Comment by psavery** [comment]
@jourdain This action currently builds the image for every merge/push to master with arm64 and amd64 architectures. It pushes the following tags to Docker Hub: `kitware/trame:latest`, `kitware/trame:ubuntu-20.04-py39`, and `kitware/trame:${{ env.MONTH_STAMP }}`, where the current month stamp tag would be `kitware/trame:02-2022`, for instance.

I **believe** the push to Docker Hub should work with the secrets, but I can't test it to confirm as the PR does not have access to the secrets (for security reasons). Maybe we can confirm that it works after merging?

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      # To emulate other architectures
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+
+      # For multi-platform builds
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Create month stamp
+        run: echo "MONTH_STAMP=$(date +%m-%Y)" >> $GITHUB_ENV
```
Comment (jourdain) [inline .github/workflows/docker.yml:26]: `+%Y-%m` instead

Comment (psavery) [reply .github/workflows/docker.yml]: Fixed

**Pull request merged**